src/Dapr.Client/DaprClientGrpc.cs

```diff
@@ -1373,6 +1376,582 @@ private Autogenerated.StateItem ToAutogeneratedStateItem(StateTransactionRequest
+        public override async IAsyncEnumerable<ReadOnlyMemory<byte>> EncryptAsync(string vaultResourceName, Stream plaintextStream,
+            string keyName, EncryptionOptions encryptionOptions, [EnumeratorCancellation] CancellationToken cancellationToken = default)
+        {
+            ArgumentVerifier.ThrowIfNullOrEmpty(vaultResourceName, nameof(vaultResourceName));
+            ArgumentVerifier.ThrowIfNullOrEmpty(keyName, nameof(keyName));
+            ArgumentVerifier.ThrowIfNull(plaintextStream, nameof(plaintextStream));
+            ArgumentVerifier.ThrowIfNull(encryptionOptions, nameof(encryptionOptions));
+
+            var shouldOmitDecryptionKeyName = string.IsNullOrWhiteSpace(encryptionOptions.DecryptionKeyName); //Whitespace isn't likely a valid key name either
+
+            var encryptRequestOptions = new Autogenerated.EncryptRequestOptions
+            {
+                ComponentName = vaultResourceName,
+                DataEncryptionCipher = encryptionOptions.EncryptionCipher.GetValueFromEnumMember(),
+                KeyName = keyName,
+                KeyWrapAlgorithm = encryptionOptions.KeyWrapAlgorithm.GetValueFromEnumMember(),
+                OmitDecryptionKeyName = shouldOmitDecryptionKeyName
+            };
+
+            if (!shouldOmitDecryptionKeyName)
+            {
+                ArgumentVerifier.ThrowIfNullOrEmpty(encryptionOptions.DecryptionKeyName, nameof(encryptionOptions.DecryptionKeyName));
+                encryptRequestOptions.DecryptionKeyName = encryptRequestOptions.DecryptionKeyName;
+            }
+
+            var options = CreateCallOptions(headers: null, cancellationToken);
+            var duplexStream = client.EncryptAlpha1(options);
+
+            //Run both operations at the same time, but return the output of the streaming values coming from the operation
+            var receiveResult = Task.FromResult(RetrieveEncryptedStreamAsync(duplexStream, cancellationToken)); 
+            await Task.WhenAll(
+                //Stream the plaintext data to the sidecar in chunks
+                Task.FromResult(SendPlaintextStreamAsync(plaintextStream, encryptionOptions.StreamingBlockSizeInBytes,
+                    duplexStream, encryptRequestOptions, cancellationToken)),
+                //At the same time, retrieve the encrypted response from the sidecar
+                receiveResult);
+            
+            //Return only the result of the `RetrieveEncryptedStreamAsync` method
+            await foreach (var item in receiveResult.Result.WithCancellation(cancellationToken))
+            {
+                yield return item;
+            }
+        }
+
+        /// <summary>
+        /// Sends the plaintext bytes in chunks to the sidecar to be encrypted.
+        /// </summary>
+        private async IAsyncEnumerable<byte[]> SendPlaintextStreamAsync(Stream plaintextStream,
```

Comment:
Can this be made to return `Task` rather than an (unused) async enumerable?

Reply:
It certainly can. Done.